core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js

```diff
@@ -1,63 +1,236 @@
+import { Button, FormTextarea, Switch, MessageStrip } from 'fundamental-react';
+import * as jp from 'jsonpath';
 
-import { CreateForm } from 'shared/components/CreateForm/CreateForm';
+import { ResourceForm } from 'shared/ResourceForm/ResourceForm';
+import * as Inputs from 'shared/ResourceForm/components/Inputs';
+import { base64Decode, base64Encode } from 'shared/helpers';
+import { IssuerRef } from 'shared/components/ResourceRef/IssuerRef';
+import { SecretRef } from 'shared/components/ResourceRef/SecretRef';
 
-import { toYaml, fromYaml, createTemplate } from './helpers';
+import { createTemplate } from './templates';
 
-import { SimpleForm } from './SimpleForm';
-import { AdvancedForm } from './AdvancedForm';
+import './CreateCertificate.scss';
 
 export function CertificatesCreate({ onChange, formElementRef, namespace }) {
   const { t } = useTranslation();
-  const notification = useNotification();
-  const postRequest = usePost();
 
   const [certificate, setCertificate] = useState(createTemplate(namespace));
+  const [withCSR, setWithCSR] = useState(false);
+  const [existingSecret, setExistingSecret] = useState(false);
+  const [csrIsEncoded, setCsrIsEncoded] = useState(false);
+  const [decodeError, setDecodeError] = useState(null);
 
-  const createCertificate = async () => {
-    try {
-      await postRequest(
-        `/apis/cert.gardener.cloud/v1alpha1/namespaces/${namespace}/certificates/`,
-        toYaml(certificate),
-      );
-      notification.notifySuccess({
-        content: t('certificates.create.messages.success'),
-      });
-      LuigiClient.linkManager()
-        .fromContext('namespace')
-        .navigate(`/certificates/details/${certificate.name}`);
-    } catch (e) {
-      console.error(e);
-      notification.notifyError({
-        title: t('certificates.create.messages.failure'),
-        content: e.message,
-      });
-      return false;
+  const csrValue = value => {
+    if (csrIsEncoded) {
+      return value;
+    } else {
+      try {
+        setDecodeError(null);
+        return base64Decode(value || '');
+      } catch (e) {
+        setDecodeError(e.message);
+        setCsrIsEncoded(true);
+        return '';
+      }
     }
   };
 
+  useEffect(() => {
+    if (withCSR) {
+      jp.value(certificate, '$.spec.commonName', undefined);
+      jp.value(certificate, '$.spec.dnsNames', undefined);
+    } else {
+      jp.value(certificate, '$.spec.csr', undefined);
+    }
+    setCertificate(certificate);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [withCSR]);
+
+  useEffect(() => {
+    if (existingSecret) {
+      jp.value(certificate, '$.spec.secretRef.name', undefined);
+    } else {
+      jp.value(certificate, '$.spec.secretRef', undefined);
+    }
+    setCertificate(certificate);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [existingSecret]);
+
+  useEffect(() => {
+    setWithCSR(!!jp.value(certificate, '$.spec.csr'));
```

Comment:
If I start typing the `CSR` value and then delete the whole string, the CSR toggle is set to false.